gluonnlp/data/__init__.py

```diff
@@ -32,6 +32,8 @@
 
 from .dataset import *
 
+# Needed for izip / zip in stream.py.
+# pylint: disable=redefined-builtin
```

Comment:
this should goes to the beginning of the stream.py
BTW, I don't think it's a good idea to override zip with izip. Because some other guy may not follow it.

Reply:
I would also prefer to put`pylint: disable=redefined-builtin` only in stream.py, but pylint continues complaining until it is added in __init__.py. I am not sure why.

I don't think there is any use-case for the Python2 zip (which returns lists instead of iterables), as anyways all code in GluonNLP must support Python3. As we are working with streams here it is crucial that zip returns an iterable and not a list. The python3 version of zip is called izip in python 2.